### PR TITLE
051 singular matrix error

### LINKED ISSUE #51: Singular matrix error in mixed convection solver
@Nino-ndrk reported an issue related to a singular matrix error occurring in the mixed convection solver. The issue is observed when prescribing a zero-power inlet condition.

The reported error message is:

```
 File "/dassh/region_mixed.py", line 269, in _solve_system
    xx = np.linalg.solve(AA, bb)
  File "/numpy/linalg/_linalg.py", line 410, in solve
    r = gufunc(a, b, signature=signature)
  File "/numpy/linalg/_linalg.py", line 104, in _raise_linalgerror_singular
    raise LinAlgError("Singular matrix")
numpy.linalg.LinAlgError: Singular matrix
```

### PR DESCRIPTION
**Pull request by fpepe-polito**
An issue related to a singular matrix error occurring in the mixed convection solver was reported by @Nino-ndrk.

The detailed explanation of the bug and the approach used to resolve it can be found in https://github.com/newcleo-dev-team/dassh_nc/issues/51#issuecomment-5003635919.

### CODE REVIEW

File: dassh/region_mixed.py [hunk 1]
```diff
@@ -364,7 +366,9 @@ def _build_vector(self, qq: np.ndarray, dz: float, z: float,
              self.params['de'][self.subchannel.type[:nn]])
         # Build energy terms of the known vector
         energy_b = qq * dz / self.params['area'][self.subchannel.type[:nn]] \
-            + EEX
+            + EEX + self._hstar * ((self._sc_vel + self._delta_v)
+                                   *self._delta_rho + 
+                                   self.sc_properties['density']*self._delta_v)
```
Comment (lelaus) [inline dassh/region_mixed.py]: Please, apply PEP8 style guide.

File: dassh/region_mixed.py [hunk 2]
```diff
@@ -572,22 +576,32 @@ def _build_matrix(self, dz: float, delta_v: np.ndarray,
         return AA
 
 
-    def _calc_h_v_star(self, delta_v: np.ndarray, delta_rho: np.ndarray, 
-                       RR: np.ndarray, nn: int) -> None:
+    def _calc_star_quantity(self, delta_v: np.ndarray, delta_rho: np.ndarray, 
+                            nn: int, variable: str, 
+                            RR: np.ndarray = None) -> np.ndarray:
```
Comment (lelaus) [inline dassh/region_mixed.py]: Please, add `None` to the type hint of the argument `RR`.

File: dassh/region_mixed.py [hunk 3]
```diff
@@ -572,22 +576,32 @@ def _build_matrix(self, dz: float, delta_v: np.ndarray,
...
             Variation of the SC densities (kg/m^3)
-        RR : np.ndarray
-            Enthalpy variation coefficient (J*m^3/kg^2)
         nn : int
             Number of coolant subchannels
-            
+        variable : str
+            Indicate whether to calculate hstar or vstar; 
+            options are 'h' or 'v'
+        RR : np.ndarray
```
Comment (lelaus) [inline dassh/region_mixed.py]: `RR` is an optional argument. Please, add ", optional" to its type hint in the docstring.

File: dassh/region_mixed.py [hunk 4]
```diff
@@ -572,22 +576,32 @@ def _build_matrix(self, dz: float, delta_v: np.ndarray,
...
+            Enthalpy variation coefficient (J*m^3/kg^2);
```
Comment (lelaus) [inline dassh/region_mixed.py]: I suggest changing the description of `RR` with:
"Derivative of enthalpy with respect to density."

File: dassh/region_mixed.py [hunk 5]
```diff
@@ -617,7 +632,6 @@ def _calc_h_v_star(self, delta_v: np.ndarray, delta_rho: np.ndarray,
         # Iterate over subchannels and adjacent subchannels
         for i in range(nn):
             denominator = 0.0
-            num_h = 0.0
             num_v = 0.0
```
Comment (lelaus) [inline dassh/region_mixed.py]: To improve readability, I suggest changing the name of the variable `num_v` with `num` to be more generic. The same goes for the variable `numerator_v`, which I suggest replacing with `numerator`.

File: tests/test_region_mixed.py
Comment (lelaus) [inline tests/test_region_mixed.py:1]: I suggest adding a test to check that the simulation runs without issues when the imposed power at the inlet is zero. In this case zero temperature increase should be obtained.

**Review by lelaus — CHANGES_REQUESTED** [review]

File: dassh/region_mixed.py (hunk 1 above)
Comment (fpepe-polito) [reply dassh/region_mixed.py]: Managed in 26b4d30.

File: dassh/region_mixed.py (hunk 2 above)
Comment (fpepe-polito) [reply dassh/region_mixed.py]: Managed in 26b4d30.

File: dassh/region_mixed.py (hunk 3 above)
Comment (fpepe-polito) [reply dassh/region_mixed.py]: Managed in 26b4d30.

File: dassh/region_mixed.py (hunk 4 above)
Comment (fpepe-polito) [reply dassh/region_mixed.py]: Managed in 26b4d30.

File: dassh/region_mixed.py (hunk 5 above)
Comment (fpepe-polito) [reply dassh/region_mixed.py]: Managed in 26b4d30.

File: tests/test_region_mixed.py
Comment (fpepe-polito) [reply tests/test_region_mixed.py:1]: Managed in 26b4d30 and 97e1c74.

**Review by lelaus — APPROVED** [review]
All my comments have been managed satisfactorily.
